### PR TITLE
Ruby Depsolver - Add `msg` to Erchef Logs

### PR DESCRIPTION
**Pull request by sdelano**
This pull request updates the temp/ruby-depsolver branch with changes from master.

I've verified that the changes (with the necessary configuration) result in `msg` showing up in the logs.

/cc @seth @hosh

### CODE REVIEW

**Comment by seth** [comment]
I say: :ship: it

Do we need some additional code to put something useful in msg for the ruby depsolver stuff?

**Comment by sdelano** [comment]
The `handle_depsolver_result` function still sets the message:

https://github.com/opscode/chef_wm/blob/temp/ruby-depsolver/src/chef_wm_depsolver.erl#L226-L256

The only path that currently doesn't set the message is during Ruby exceptions. I can fix that up according to comments from the last pull request.

File: src/chef_wm_depsolver.erl [hunk 1]
```diff
@@ -243,7 +239,7 @@ handle_depsolver_results(ok, {error, exception, Message, Backtrace}, Req, State)
                                {error_type, ruby_exception},
                                {message, Message},
                                {backtrace, Backtrace}]),
-    {{halt, 500}, Req, State};
+    {{halt, 500}, Req, State#base_state{log_msg = ruby_exception};
```
Comment (seth) [inline src/chef_wm_depsolver.erl]: maybe worth using the local `server_error` helper function here?

**Comment by seth** [comment]
Otherwise, looks good. :+1:

File: src/chef_wm_depsolver.erl (hunk 1 above)
Comment (sdelano) [inline src/chef_wm_depsolver.erl]: Updated.

**Comment by seth** [comment]
Since we're (or you're) here tweaking this code, I think we should standardize the log_msgs just a bit. We have:

```
solve_timeout
no_depsolver_workers
ruby_exception
not_reachable_dep
invalid_constraints
cookbook_version_not_found
{timeout, cookbook_authz}
{read, forbidden}
```

Since we capture perf logs under `depsolver_time` and `depsolver_count`, I'd prefer we standardize on that:

```
{timeout, depsolver} (matches cookbook_authz) or depsolver_timeout
unreachable_dep
depsolver_exception
(unrelated, but {forbidden, read} would seem a better order since it generalizes better for other perms)
```

Here are a couple of alternatives for the error messages:

```
<<"Out of depednecy solvers. Try again later.">>
<<"Server overloaded, try again later.">>
<<"Dependency solver overloaded, try again later.">>

<<"Dependency solving failed.">>
<<"Dependency solver failed.">>
<<"Dependency solver exception.">>
```

**Comment by sdelano** [comment]
Ping @manderson26 @marcparadise to see comments on standardizing error messages.

**Pull request merged**